sdcm/nemesis_registry.py

```diff
@@ -0,0 +1,161 @@
+import ast
+import inspect
+import re
+from functools import lru_cache
+from typing import List, Optional, TypeVar, Callable
+
+from sdcm.utils.ast_utils import BooleanEvaluator
+
+DISRUPT_PATTERN = re.compile(r"self\.(?P<method_name>disrupt_[0-9A-Za-z_]+?)\(.*\)", flags=re.MULTILINE)
+SourceType = TypeVar("SourceType")
+DisruptMethod = Callable[[], None]
+
+
+@lru_cache
+def get_disrupt_method_from_class(nemesis_cls):
+    method_name = DISRUPT_PATTERN.search(inspect.getsource(nemesis_cls.disrupt))
+    if method_name:
+        return method_name.group("method_name")
+
+
+class NemesisRegistry:
+    def __init__(self, base_class: SourceType, kubernetes=False, excluded_list=None):
+        super().__init__()
+        self.base_class = base_class
+        self.kubernetes = kubernetes
```

Comment:
I would take this out, if we want to filter this by default when kubernetes is used, let's add the where we read the filter

it's not a first class variable of this registry, it just one of the flags from it's concern.